tests/components/tuya/snapshots/test_sensor.ambr

```diff
@@ -1783,6 +1783,58 @@
     'state': '0.0',
   })
 # ---
+# name: test_platform_setup_and_discovery[sensor.hl400_pm2_5-entry]
+  EntityRegistryEntrySnapshot({
+    'aliases': set({
+    }),
+    'area_id': None,
+    'capabilities': dict({
+      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
+    }),
+    'config_entry_id': <ANY>,
+    'config_subentry_id': <ANY>,
+    'device_class': None,
+    'device_id': <ANY>,
+    'disabled_by': None,
+    'domain': 'sensor',
+    'entity_category': None,
+    'entity_id': 'sensor.hl400_pm2_5',
+    'has_entity_name': True,
+    'hidden_by': None,
+    'icon': None,
+    'id': <ANY>,
+    'labels': set({
+    }),
+    'name': None,
+    'options': dict({
+    }),
+    'original_device_class': None,
+    'original_icon': None,
+    'original_name': 'PM2.5',
+    'platform': 'tuya',
+    'previous_unique_id': None,
+    'suggested_object_id': None,
+    'supported_features': 0,
+    'translation_key': 'pm25',
+    'unique_id': 'tuya.zfHZQ7tZUBxAWjACjkpm25',
+    'unit_of_measurement': '',
```

Comment:
empty UoM? That sounds incorrect

Reply:
Would it make sense to set an empty string to `None`?

Reply:
> Would it make sense to set an empty string to `None`?

This fix reverts to the previous behavior (<= `2025.7`)

Empty string is not technically an invalid value when the device class is not set, so I think that should be considered in a separate PR (and maybe on the base sensor platform instead of every integration?).